CodeEdit/CodeEditApp.swift

```diff
@@ -30,9 +34,19 @@ struct CodeEditApp: App {
             AboutWindow()
 
             SettingsWindow()
-                .commands {
-                    CodeEditCommands()
-                }
+
+            NSDocumentGroup(for: WorkspaceDocument.self) { workspace in
+                WindowSplitView(workspace: workspace)
+            } defaultAction: {
+                openWindow(id: SceneID.welcome.rawValue)
+            }
+            .register(.document(WorkspaceDocument.self))
+            .transition(.documentWindow)
+            .windowToolbarStyle(.unifiedCompact(showsTitle: false))
+            .enableOpenWindow()
+            .commands {
+                CodeEditCommands()
+            }
```

Comment:
It's not readable to me what is happening here. Can the naming be improved or it just not possible to improve that?

Reply:
These things are from [this](https://github.com/Wouter01/SwiftUI-WindowManagement) package and should all have documentation attached to them. To achieve the desired functionality, some modifiers like `register` and `enableOpenWindow` need to be used which can be a bit confusing, but the docs mention this. `NSDocumentGroup` is a replacement for [DocumentGroup](https://developer.apple.com/documentation/swiftui/documentgroup) and has a similar API. The defaultAction is called when the user clicks on the "+" button when window tabbing is enabled, and will open the CodeEdit Welcome Window in this case. Does this make it more clear?

Reply:
I added some comments to clarify the use